src/plugins/vis_type_timeline/server/lib/fetch_data_source_id.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObjectsClientContract } from 'src/core/server';
+import { DataSourceAttributes } from 'src/plugins/data_source/common/data_sources';
+import { getDataSourceEnabled } from './services';
+import { OpenSearchFunctionConfig } from '../types';
+
+export const fetchDataSourceIdByName = async (
+  config: OpenSearchFunctionConfig,
+  client: SavedObjectsClientContract
+) => {
+  if (!config.data_source_name) {
+    return undefined;
+  }
+
+  if (!getDataSourceEnabled().enabled) {
```

Comment:
I can imagine a world where saved objects were imported from a cluster that enabled MDS to a new domain that didn't enable it.

Do we want to consider that if it's not enabled it just queries to the local cluster?

Reply:
Existing behavior with Timeline is that any invalid params are caught and a toast error is displayed, meaning before this merge, if a user types in `data_source_name`, a toast would be thrown anyway because this was not a named argument. This means that any syntax error or invalid param can and should be caught. If the visualization just queries a local cluster, then it can be confusing behavior especially since this change enables Timeline to query both local cluster and MDS queries within one visualization.